lang/src/encode_input_data.rs

```diff
@@ -0,0 +1,78 @@
+use ink_core::memory::vec::Vec;
+use ink_utils;
+use parity_codec::Encode;
+
+pub trait EncodeSafe{
```

Comment:
What is the point?

Reply:
Since parity_codec::Encode includes generic type, cannot make trait object of "Encode". So i temporary made another trait to use in this code, which is not using generic types.

Reply:
It is not idiomatic to do that and confusing. Please find another way to test this functionality. You are free to make limited use of macro_rules! macros in order to provide the functionality of your tests if required.